geoapps/utils/utils.py

```diff
@@ -40,6 +41,37 @@
 from sklearn.neighbors import KernelDensity
 
 
+def get_locations(workspace: Workspace, object: UUID | Entity):
+    """
+    Returns object's centroids or vertices.
+
+    If no location data is found on the provided object, the method will
+    attempt to call itself on it's parent.
+
+    :param workspace: Geoh5py Workspace object.
+    :param object: Object or uuid of object containing centroid or
+        vertex location data.
+
+    :return: Array shape(*, 3) of x, y, z location data
+
+    """
+
+    if isinstance(object, UUID):
+        object = workspace.get_entity(object)[0]
```

Comment:
This call could return None if the UUID isn't in the workspace, which will make Line 67 to crash without proper error. See suggestion below.